docs/ocean/design_docs/min_level_cell.rst

```diff
@@ -0,0 +1,729 @@
+
+  * [ ] ``ocn_time_integration_split_init()``
+
+  * [ ] ``ocn_time_integrator_split()``
+
+* [ ] ``ocn_mesh``:
+
+  * [ ] ``minLevelCell``
+
+  * [ ] ``minLevelEdgeTop``
+
+  * [ ] ``minLevelEdgeBot``
+
+  * [ ] ``minLevelVertexTop``
+
+  * [ ] ``minLevelVertexBot``
+
+  * [ ] ``ocn_meshCreate()``
+
+  * [ ] ``ocn_meshUpdateFields()``
+
+  * [ ] ``ocn_meshDestroy()``
+
+* [ ] ``ocn_tendency``:
+  
+  * [ ] ``ocn_tend_tracer``
+
+* [ ] ``ocn_forcing``:
+
+  * [ ] ``ocn_forcing_build_fraction_absorbed_array``
+
+* [ ] ``ocn_thick_hadv``:
+  
+  * [ ] ``ocn_hadv_thick_tend``
+  
+* [ ] ``ocn_thick_vadv``:
+  
+  * [ ] ``ocn_vadv_thick_tend``
+
+* [ ] ``ocn_thick_surface_flux``:
+  
+  * [ ] ``ocn_thick_surface_flux_tend``
+  
+* [ ] ``ocn_tracer_advection``:
+  
+  * [ ] ``ocn_tracer_advection_tend``
+
+* [ ] ``ocn_tracer_advection_mono``:
+  
+  * [ ] ``ocn_tracer_advection_mono_tend``
+
+* [ ] ``ocn_vel_forcing_surface_stress``:
+  
+  * [ ] ``ocn_vel_forcing_surface_stress_tend``
+  
+* [ ] ``ocn_vel_hmix_del2``:
+  
+  * [ ] ``ocn_vel_hmix_del2_tend``
+  
+* [ ] ``ocn_vel_hadv_coriolis``:
+  
+  * [ ] ``ocn_vel_hadv_coriolis_tend``
+
+* [ ] ``ocn_vel_vadv``:
+  
+  * [ ] ``ocn_vel_vadv_tend``
+
+* [ ] ``ocn_vmix_cvmix``:
+  
+  * [ ] ``ocn_vmix_coefs_cvmix_build``
+
+* [ ] ``ocn_diagnostics``:
+
+  * [ ] ``ocn_relativeVorticity_circulation()``
+
+  * [ ] ``ocn_diagnostic_solve_layerThicknessEdge()``
+
+  * [ ] ``ocn_diagnostic_solve_vorticity()``
+
+  * [ ] ``ocn_diagnostic_solve_richardson()``
+
+  * [ ] ``ocn_diagnostic_solve_surfaceLayer()``
+
+  * [ ] ``ocn_diagnostic_solve_vortVel()``
+
+  * [ ] ``ocn_diagnostic_solve_z_coordinates()``
+
+  * [ ] ``ocn_diagnostic_solve_pressure()``
+
+  * [ ] ``ocn_vert_transport_velocity_top()``
+
+  * [ ] ``ocn_fuperp()``
+
+  * [ ] ``ocn_compute_land_ice_flux_input_fields()``
+
+  * [ ] ``ocn_validate_state()``
+
+* [ ] ``ocn_thick_ale``:
+
+  * [ ] ``ocn_ALE_thickness``
+
+* [ ] ``ocn_vel_pressure_grad``:
+  
+  * [ ] ``ocn_vel_pressure_grad_tend``
+
+* [ ] ``ocn_vmix``:
+  
+  * [ ] ``ocn_vmix_implicit``
+  
+  * [ ] ``ocn_vel_vmix_tend_implicit``
+  
+  * [ ] ``ocn_tracer_vmix_tend_implicit``
+
+...
+
+.. note::
+
+  ``ocn_equation_of_state_jm`` currently doesn't include any reference to
+  ``maxLevel*`` but this is mildly concerning.  T and S are clipped to the
+  valid range before density is computed, meaning that contamination with
+  invalid values could go unnoticed.  Still, it does appear that
+  ``displacedDensity`` is currently only used starting at index ``k = 2``, so
+  no invalid values should be getting used.
+
+.. note::
+
+  ``ocn_forcing_build_fraction_absorbed_array`` is currently only called once 
+  by ``ocn_init_routines`` and would need to be called multiple times to correctly
+  distribute surface fluxes unless we use an alternative approach where the vertical
+  index of transmissionCoeff is number of cells from minLevelCell rather than k-levels
```

Comment:
@cbegeman, I'm seeing this function getting called every time step here:
https://github.com/MPAS-Dev/MPAS-Model/blob/ocean/develop/src/core_ocean/mode_forward/mpas_ocn_forward_mode.F#L574
So I think even in the future when `minLevelCell` changes in time, this should work.  Did I misunderstand something?

Also, I think we would definitely want to make the transmission coefficient 100% (absorption 0) for cells above the first valid cell because we use this infrastructure for fluxes under ice shelves.

Reply:
@xylar Thanks for catching that. So it looks like we'd be fine on that front. 

"Transmitting" the fluxes through the inactive cells sounds like a good safeguard, though if we change the loop limits in `ocn_thick_surface_flux_tend` we shouldn't need to use transmission coefficient values above minLevelCell.